### PR TITLE
Add external predicate to matchEquals and matchAny in ReceiverBuilder, #25774

### PR DESCRIPTION
**Pull request by patriknw**
For discussion if it should be included and then it could be merged after 2.6.0.

Original was in https://github.com/akka/akka/pull/25822

### CODE REVIEW

**Comment by hepin1989** [comment]
Totally fine, let's do this after release 2.6.0. Make Java API more complete is always great.

**Comment by hepin1989** [comment]
Will this be included in the next release?

File: akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java [hunk 1]
```diff
@@ -240,22 +239,50 @@ public boolean defined(Object o) {
...
+   * @param object the object to compare equals with
+   * @param externalPredicate an external predicate that will be evaluated if the object compares
+   *     equal
+   * @param apply an action to apply to the argument if the object compares equal
+   * @return a builder with the case statement added
+   */
+  @SuppressWarnings("unchecked")
+  public <P> ReceiveBuilder matchEquals(
+      final P object,
+      final java.util.function.BooleanSupplier externalPredicate,
```
Comment (patriknw) [inline akka-actor/src/main/java/akka/japi/pf/ReceiveBuilder.java]: I don't see much value of having the overload, but we have `BooleanSupplier` in for example `match` so I guess it's good to add here also to make it consistent.

The syntactic difference is `notUsed -> !nodes.isEmpty()` vs `() -> !nodes.isEmpty()`.

**Review by patriknw — COMMENTED** [review]
LGTM, because we already have it for the other things (but I don't think it adds much value and we shouldn't add such to the Typed builders)

**Review by jrudolph — COMMENTED** [review]
I agree with @patriknw, I'm not sure if it adds much value but it might not hurt. It needs tests.

**Review by helena — APPROVED** [review]
Looks ok.

**Pull request merged**